tests/ballerina-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java

```diff
@@ -503,129 +606,181 @@ public void testDateTimeOutParams() {
+        BValue[] returns = BRunUtil.invoke(resultNegative, "testSelectData", connectionArgs);
         Assert.assertTrue(returns[0].stringValue().contains("execute query failed:"));
     }
 
     @Test(groups = "ConnectorTest", description = "Test failed update with generated id action")
     public void testFailedGeneratedKeyOnInsert() {
-        BValue[] returns = BRunUtil.invoke(resultNegative, "testGeneratedKeyOnInsert");
+        BValue[] returns = BRunUtil.invoke(resultNegative, "testGeneratedKeyOnInsert", connectionArgs);
         Assert.assertTrue(returns[0].stringValue().contains("execute update with generated keys failed:"));
     }
 
     @Test(groups = "ConnectorTest", description = "Test failed call procedure")
     public void testFailedCallProcedure() {
-        BValue[] returns = BRunUtil.invoke(resultNegative, "testCallProcedure");
+        BValue[] returns = BRunUtil.invoke(resultNegative, "testCallProcedure", connectionArgs);
         Assert.assertTrue(returns[0].stringValue().contains("execute stored procedure failed:"));
     }
 
     @Test(groups = "ConnectorTest", description = "Test failed batch update")
     public void testFailedBatchUpdate() {
-        BValue[] returns = BRunUtil.invoke(resultNegative, "testBatchUpdate");
-        Assert.assertTrue(returns[0].stringValue().contains("execute batch update failed:"));
+        BValue[] returns = BRunUtil.invoke(resultNegative, "testBatchUpdate", connectionArgs);
+        if (dbType == HSQLDB) {
+            Assert.assertTrue(returns[0].stringValue().contains("execute batch update failed:"));
+        } else {
+            Assert.assertTrue(returns[0].stringValue().contains("failure"));
+        }
     }
 
     @Test(groups = "ConnectorTest", description = "Test failed parameter array update")
     public void testInvalidArrayofQueryParameters() {
-        BValue[] returns = BRunUtil.invoke(resultNegative, "testInvalidArrayofQueryParameters");
+        BValue[] returns = BRunUtil.invoke(resultNegative, "testInvalidArrayofQueryParameters", connectionArgs);
         Assert.assertTrue(returns[0].stringValue()
                 .contains("execute query failed: unsupported array type for parameter index 0"));
     }
 
     @Test(groups = "ConnectorTest", description = "Test failure scenario in adding data to mirrored table")
     public void testAddToMirrorTableNegative() throws Exception {
-        BValue[] returns = BRunUtil.invoke(resultMirror, "testAddToMirrorTableNegative");
-        Assert.assertNotNull(returns);
-        Assert.assertEquals(returns[0].stringValue(), "{message:\"execute update failed: integrity constraint "
+        BValue[] returns = BRunUtil.invoke(resultMirror, "testAddToMirrorTableNegative", connectionArgs);
+        String errorMessage = "{message:\"execute update failed: integrity constraint "
                 + "violation: unique constraint or index violation; SYS_PK_10179 table: EMPLOYEEADDNEGATIVE\", "
```

Comment:
Is this "SYS_PK_10179 " is an auto generated id for the constrained? Can this be changed

Reply:
Probably this doesn't change since the order of table creation never changes. But better if we remove this.

Reply:
But order of table creating can be changed based on several factors. So better we avoid this